plugins/opensslcryptoplugin/opensslcryptoplugin.cpp

```diff
@@ -18,20 +20,83 @@
 #include <QtCore/QUuid>
 #include <QtCore/QCryptographicHash>
 
+#include <fstream>
+#include <cstdlib>
+
+#include <openssl/rand.h>
+
 Q_PLUGIN_METADATA(IID Sailfish_Crypto_CryptoPlugin_IID)
 
 using namespace Sailfish::Crypto;
 
 Daemon::Plugins::OpenSslCryptoPlugin::OpenSslCryptoPlugin(QObject *parent)
     : QObject(parent), CryptoPlugin()
 {
+    // seed the RNG
+    char seed[16] = {0};
+    std::ifstream rand("/dev/urandom");
+    rand.read(seed, 16);
+    rand.close();
+    RAND_add(seed, 16, 1.0);
+
+    // initialise EVP
     osslevp_init();
 }
 
 Daemon::Plugins::OpenSslCryptoPlugin::~OpenSslCryptoPlugin()
 {
 }
 
+Result
+Daemon::Plugins::OpenSslCryptoPlugin::generateRandomData(
+        quint64 callerIdent,
+        const QString &csprngEngineName,
+        quint64 numberBytes,
+        QByteArray *randomData)
+{
+    Q_UNUSED(callerIdent)
+
+    if (csprngEngineName != GenerateRandomDataRequest::DefaultCsprngEngineName) {
+        return Result(Result::CryptoPluginRandomDataError,
+                      QLatin1String("The OpenSSL crypto plugin doesn't currently support other RNG engines")); // TODO!
+    }
+
+    if (numberBytes >= 4097 || numberBytes <= 0) {
+        return Result(Result::CryptoPluginRandomDataError,
+                      QLatin1String("The OpenSSL crypto plugin can only generate up to 4096 bytes of random data at a time"));
+    }
+
+    int nbytes = numberBytes;
+    unsigned char buf[4096];
```

Comment:
I'd put this `4096` and `4097` above into a single named constant.

Reply:
Yep, will use the MAX_NUMBER_OF_BYTES define as you suggested